File: app/src/main/java/com/OdiousPanda/rainbow/Utilities/FoodIconUtil.java
package com.OdiousPanda.rainbow.Utilities;

import com.OdiousPanda.rainbow.R;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class FoodIconUtil {
    private List<Field> foodIcons = new ArrayList<>();

    public FoodIconUtil() {
        Field[] drawablesFields = com.OdiousPanda.rainbow.R.drawable.class.getFields();
        for (Field field : drawablesFields) {
            try {
                if (field.getName().contains("dish") || field.getName().contains("dessert") || field.getName().contains("drink")) {
                    foodIcons.add(field);
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    public int getRandomIcons() {
        try {
            return foodIcons.get(new Random().nextInt(foodIcons.size())).getInt(null);
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        return R.drawable.dish_ic_hamburger;
    }

}

File: app/src/main/java/com/odiousPanda/rainbowKt/utilities/FoodIconUtil.kt
package com.odiousPanda.rainbowKt.utilities

import com.odiousPanda.rainbowKt.R
import java.lang.reflect.Field
import java.util.*

class FoodIconUtil {
    private val foodIcons = mutableListOf<Field>()

    init {
        val drawablesFields: Array<Field> = R.drawable::class.java.fields
        for (field in drawablesFields) {
            try {
                if (field.name.contains("dish") || field.name
                        .contains("dessert") || field.name.contains("drink")
                ) {
                    foodIcons.add(field)
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    fun getRandomIcons(): Int {
        try {
            return foodIcons[Random().nextInt(foodIcons.size)].getInt(null)
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        return R.drawable.dish_ic_hamburger
    }
}